### ISSUE TITLE
How to create an adhoc sandbox with a custom Dockerfile?

### ISSUE THREAD

**Issue by punkpeye** [codesandbox/codesandbox-sdk#11]
I have tried:

```ts
const sdk = new CodeSandbox(CODESANDBOX_API_KEY);
const sandbox = await sdk.sandbox.create({
  hibernationTimeoutSeconds: 300,
});

await sandbox.fs.writeTextFile(
  '.devcontainer/Dockerfile',
  'FROM python:3.10\nENV FOO=BAR\n',
);

await sandbox.fs.writeTextFile(
  '.devcontainer/devcontainer.json',
  stringifyJson({
    build: {
      dockerfile: './Dockerfile',
    },
    name: 'Test',
  }),
);


const result = await sandbox.shells.run('env | grep FOO');
```

**Comment by punkpeye** [comment]
Based on my reading of the documentation, I would expect that anything that's run in shell, will execute inside the Dockerfile, but that does not seem to be the case.

```
await sandbox.fs.writeTextFile(
  '.devcontainer/Dockerfile',
  multiline`
    FROM python:3.11  
    RUN echo 'Hello, World!' > index.html
  `,
);

await sandbox.fs.writeTextFile(
  '.devcontainer/devcontainer.json',
  stringifyJson({
    build: {
      dockerfile: './Dockerfile',
    },
    name: 'Test',
  }),
);

console.log(await sandbox.shells.run('ls -lah .'));
```

This returns the sandbox contents as opposed to Docker's shell.

In short, I want to deploy a service using Dockerfile and access it through a port.

**Comment by CompuIves** [comment]
Heya! What you're sharing should work, but it requires a restart of the sandbox to apply the dockerfile (`sandbox.reboot`). I would recommend to use the snapshot builder (https://codesandbox.io/docs/sdk/snapshot-builder) to precreate a template that has the Docker image loaded, and use that template for creating new sandboxes.

**Comment by punkpeye** [comment]
I eventually figured out that if I restart it works, but the restart time was way too slow for my use case.

Is snapshot something that can be built/deployed faster than using this and then doing restart?

**Comment by CompuIves** [comment]
Yes, absolutely. I would recommend creating a snapshot with the docker image, and then using that when creating new sandboxes. That will allow you to create new sandboxes with that docker image in ~1s, instead of having to wait for the full reboot.

**Comment by christianalfoni** [comment]
Closing this for now, will highlight better in docs the snapshot builder option
